lib/rack/session/redis.rb

```diff
@@ -45,10 +45,45 @@ def find_session(req, sid)
       end
 
       def write_session(req, session_id, new_session, options)
-        with_lock(req, false) do
-          with { |c| c.set session_id, new_session, options }
-          session_id
+        if threadsafe?
+          return transactional_write_session(req, session_id, new_session, options)
         end
+
+        with { |c| c.set session_id, new_session, options }
+        session_id
+      end
+
+      def transactional_write_session(req, session_id, new_session, options)
+        with { |c|
+          # Atomically read the old session and merge it with the new session
+          # Inspired by this: https://stackoverflow.com/a/11311126/7816
```

Comment:
these comments are pretty much only visible to people who read the source. is this information that users of the gem will need to know? if not, i don't see a reason for it in the source. if so, it should be in YARDoc.

Reply:
People who read and edit the code are exactly who that comment is intended for. Unless someone is already familiar with how to perform a sequence of commands atomically in Redis (I wasn't), it's not apparent what the `loop { c.watch ..., break if c.multi { ... } }` pattern accomplishes, which is why I thought a comment was in order.

I suppose I could write it as YARDoc for the `#transactional_write_session` method, but the reason I didn't, is that the intention of the comment was to provide a hint to anyone who couldn't understand the method's implementation, not to explain its purpose, which I felt was sufficiently clear from its name. Also, it would have felt a little weird to add YARDoc for that one method (which should probably have been private), when no other identifier in the project has been documented in that way.

Reply:
Can't argue with that, I'll definitely gonna do a pass on these gems at some point in the future and make sure their RDoc is up-to-date across the board. No worries.